testnet/stacks-node/src/tests/epoch_21.rs

```diff
@@ -4677,6 +4679,131 @@ fn test_sortition_divergence_pre_21() {
     }
 }
 
+#[test]
+#[ignore]
+fn test_v1_pox_default_contract() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    let reward_cycle_len = 5;
+    let prepare_phase_len = 3;
+    let epoch_2_05 = 205;
+    let epoch_2_1 = 212;
+    let v1_unlock_height = 213;
+
```

Comment:
This test looks like it could be a unit test rather than an integration test -- the `RPCPoxInfoData::from_data` function can be tested in a test like `rpc::test_rpc_getpoxinfo_use_latest_tip`. I think the `test_rpc()` function could be setup to process an epoch transition: @jcnelson, does that seem reasonable?

Reply:
I had recommended to @donpdonp to implement an integration test instead of a unit test precisely because `test_rpc()` does not execute epoch transitions, and adding that ability without breaking the other tests and duplicating code would have required more time and effort.  It was a trade-off between velocity versus code hygiene -- I prioritized velocity because it's part of the 2.1 upgrade, which is already very late.

However, if we feel that code hygiene is more important here, then yes -- creating a variant of `test_rpc()` or modifying `test_rpc()` to do an epoch transition at a pre-determined height would be the "cleaner" thing to do.

Reply:
That's fine -- in that case, this testing should be changed in two ways:

1. The assertion should just be added to an existing test: nothing is happening in this test that does not already occur in any of the other 2_1 tests
2. Because this is an end to end test, the assertion should use the `neon_integrations::get_pox_info` method rather than trying to directly read from the database

Reply:
I looked through the existing integration tests and found a good spot in test `transition_removes_pox_sunset` that was already failing due to the active contract being block-based instead of reward-cycle based. So it was a matter of updating that test to look for pox-2 at v1_unlock_height+1. The new integration test has been removed completely.